src/eureka/S5_lightcurve_fitting/s5_fit.py

```diff
@@ -410,18 +409,68 @@ def fitlc(eventlabel, ecf_path=None, s4_meta=None, input_meta=None):
                     # flux values
                     flux, flux_err = util.normalize_spectrum(meta, flux,
                                                              flux_err)
-
-                    meta, params = fit_channel(meta, time, flux, channel,
-                                               flux_err, eventlabel, params,
-                                               log, longparamlist, time_units,
-                                               paramtitles, chanrng, ld_coeffs,
-                                               xpos, ypos, xwidth, ywidth)
-
-                    # Save results
-                    log.writelog('Saving results', mute=(not meta.verbose))
-                    me.saveevent(meta, (meta.outputdir+'S5_'+meta.eventlabel +
-                                        "_Meta_Save"), save=[])
-
+                                 
+                    if hasattr(meta, 'fit_bins'):
+                        if ((channel > meta.fit_bins[0]) and 
+                           (channel < meta.fit_bins[1])):
+                            try:
+                                meta, params = fit_channel(meta, time, flux, 
+                                                           channel, flux_err, 
+                                                           eventlabel, params,
+                                                           log, longparamlist, 
+                                                           time_units, 
+                                                           paramtitles, 
+                                                           chanrng, ld_coeffs,
+                                                           xpos, ypos, 
+                                                           xwidth, ywidth)
+                            except:
+                                print(' >>> fit failed')
+                                continue
+
+                            # Save results
+                            log.writelog('Saving results', 
+                                         mute=(not meta.verbose))
+                            me.saveevent(meta, (meta.outputdir +
+                                                'S5_'+meta.eventlabel +
+                                                "_Meta_Save"), save=[])
+                        else:
+                            print(' >>> skipping bin')
+                            continue  
+                                                
+                    elif hasattr(meta, 'skip_bins'):
```

Comment:
I think there's likely a cleaner way of doing this that avoids a bunch of duplicated code and allows both `fit_bins` and `skip_bins` to be used together. Let's make a list of channels that'll be iterated through (e.g. `chan_list`) using `np.arange` and taking into account any `fit_bins` and/or `skip_bins` settings, and then use that above instead of the current `for channel in range(chanrng):`